### PR TITLE
Backend evaluator

### PR DESCRIPTION
**Pull request by teddylo57**

### CODE REVIEW

File: backend/src/main/java/poker/Api.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static void main(String[] args) {
       return handService.save(hand);
     });
 
-    get("/hands", (request, response) -> {
+    get("/hands/all", (request, response) -> {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/Api.java]: Remove `/all`?

File: backend/src/main/java/poker/Evaluator.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      ranks[i] = 0;
+    }
+    for (Card c : cardList) {
+      ranks[c.getRank().getValue()]++;
+    }
+  //    print();
+    checker(cardList);
+  }
+
+  private void checker(List<Card> cardList) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/Evaluator.java]: Let's go over this together some time so I can better understand

File: backend/src/main/java/poker/enums/RANK.java [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+  Rank(Integer value) {
+    this.value = value;
+  }
+
+  public Integer getValue() {
+    return value;
+  }
+
+  public static Rank findByValue(Integer value) {
+    for (Rank rank : values()) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/enums/RANK.java]: How come we don't use a map here?

File: backend/src/test/java/poker/DeckTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  @Test
+  public void showTest() {
+
+  }
+
+  @Test
+  public void emptyDeck() {
+    for (int i = 0; i < 36; i++) {
```
Comment (ericliu121187) [inline backend/src/test/java/poker/DeckTest.java]: I'd avoid using [magic numbers](http://wiki.c2.com/?MagicNumber). Instead, let's declare a constant `"TOTAL_CARD_COUNT"` or something and then set `36` to that.

File: backend/src/test/java/poker/EvaluatorTest.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  @Test
+  public void checkOnePair() {
+    List<Card> cardList = new ArrayList<>();
+    cardList.add(new Card(Suit.CLOVER, Rank.J));
+    cardList.add(new Card(Suit.SPADE, Rank.J));
+    cardList.add(new Card(Suit.CLOVER, Rank.A));
+    cardList.add(new Card(Suit.CLOVER, Rank.SEVEN));
+    cardList.add(new Card(Suit.CLOVER, Rank.SIX));
+    Evaluator evaluator = new Evaluator(cardList);
+    assertThat("High Card", evaluator.getResult().equals(Handrank.ONE_PAIR));
```
Comment (ericliu121187) [inline backend/src/test/java/poker/EvaluatorTest.java]: You have `"High Card"` for all of these tests. That should be different for each test, right?

Comment (teddylo57) [reply backend/src/test/java/poker/EvaluatorTest.java]: Yes I have to fix that

File: backend/src/test/java/poker/DeckTest.java (hunk 4 above)
Comment (teddylo57) [reply backend/src/test/java/poker/DeckTest.java]: Agreed it should be extracted out into a variable

File: backend/src/main/java/poker/enums/RANK.java (hunk 3 above)
Comment (teddylo57) [reply backend/src/main/java/poker/enums/RANK.java]: We could but that would be an extra step, as seen here..you'd put them into a Map and then retrieve them, we can save that step in my opinion

https://stackoverflow.com/questions/5021246/conveniently-map-between-enum-and-int-string

File: backend/src/main/java/poker/Evaluator.java (hunk 2 above)
Comment (teddylo57) [reply backend/src/main/java/poker/Evaluator.java]: Sure, the code logic will change as I've been working on finding the winning hand for 5-7 cards now.

File: backend/src/main/java/poker/Api.java (hunk 1 above)
Comment (teddylo57) [reply backend/src/main/java/poker/Api.java]: Agreed, not sure why that got merged in

**Pull request merged**